### PR TITLE
Add ability to import user-provided dSYM to apple_dynamic_framework_import

### PR DESCRIPTION
**Pull request by thii**
This adds a new `dsym_imports` attribute to
`apple_dynamic_framework_import` rule, that will allow users to import
the dSYM bundle of the imported framework. This is currently only used
for generating and packaging `.symbols` files into the `.ipa` file for
submitting to the App Store.

Closes https://github.com/bazelbuild/rules_apple/issues/892.

### CODE REVIEW

File: .bazelci/presubmit.yml
Comment (thii) [reply .bazelci/presubmit.yml:21]: I hesitated to do that since it would unnecessarily slow down the builds of targets that don't need dSYMs and symbols generated.

Comment (thii) [reply .bazelci/presubmit.yml:21]: I can give it a test run.

Comment (thii) [reply .bazelci/presubmit.yml:21]: The build fails if we pass `--apple_generate_dsym` to everything, because of output paths collision. (There's a patch that attempts to fix this: https://github.com/bazelbuild/rules_apple/pull/715)

**Comment by thii** [comment]
Updated to also extracting debug symbols from frameworks that don't provide dSYMs. Thanks @brentleyjones for letting me know this.

**Comment by brentleyjones** [comment]
@thii There is a test failiure

**Comment by thii** [comment]
It was because of an incompatible flag being flipped in https://github.com/bazelbuild/bazel/commit/213043a1b9fa5053fae720220f28ebc604283079. Fixed here: https://github.com/bazelbuild/rules_apple/pull/965.

**Comment by brentleyjones** [comment]
Merged the fix, can you rebase? Thanks!

**Review by brentleyjones — APPROVED** [review]
This is nice, thanks @thii for including the non-dSYM case.

I can't test `master` yet locally, so I'll wait for someone else to approve as well before we merge it.

**Review by brentleyjones — APPROVED** [review]
I hadn't realized that `apple.package_symbols` defaulted to `no` (our version had that flipped).

This generates all of our symbols now (though more than we had before, because we only did `arm64` before, and I see your comment on not knowing the arch yet. Is there a way to narrow it down, similar to show we strip architectures of imported frameworks? And if not, have you verified that submitting more symbols than needed doesn't cause an upload error in Appstore Connect?

File: apple/internal/partials/framework_import.bzl
Comment (thii) [inline apple/internal/partials/framework_import.bzl:101]: When debugging this PR, I've found that this may contain duplicate values, so I changed the implementation of it. Not sure if I should submit it here.

File: apple/internal/partials/framework_import.bzl
Comment (thii) [inline apple/internal/partials/framework_import.bzl:286]: Not sure if we should optimize for this case since it's very uncommon. If we do, then it might be better to extract these commands into a tool, than adding more ad-hoc bash here.

File: apple/internal/partials/framework_import.bzl
Comment (brentleyjones) [reply apple/internal/partials/framework_import.bzl:101]: > Not sure if I should submit it here.

Is the change you have here what you aren't sure of? Without the change, what impact does the duplicates have?

File: apple/internal/partials/framework_import.bzl
Comment (brentleyjones) [reply apple/internal/partials/framework_import.bzl:286]: I'm fine with not optimizing for now, that is something we can improve later. Thoughts @keith?

File: apple/internal/partials/framework_import.bzl
Comment (thii) [reply apple/internal/partials/framework_import.bzl:101]: e.g. `[x86_64, x86_64, x86_64]`

Comment (thii) [reply apple/internal/partials/framework_import.bzl:101]: The change is kind of unrelated to this PR. I can submit it separately if this PR is going to take some more time. 

Without this, some commands may end up being executed more than once.

Comment (brentleyjones) [reply apple/internal/partials/framework_import.bzl:101]: I would submit it as a followup PR, since it's unrelated to the core of the change.

**Comment by thii** [comment]
The failure with the last green Bazel seems to be because of https://github.com/bazelbuild/bazel/commit/d3fc253a49a00c34408bbaf5378376cbcea1c5c9.

**Review by brentleyjones — APPROVED** [review]

**Comment by brentleyjones** [comment]
Thanks again @thii!

**Pull request merged**
